add 2016 swag email

binary1230 opened the pull request:
now OK for merging pending review

@nickthenewbie1 wants to get this out asap, just need to bolt down the last bit and test out the HTML.

The HTML contains some hard coded dates in it which we'd need to update next year.

EliAndrewC commented on magprime/templates/emails/attendee_swag_promo.html:
```diff
@@ -0,0 +1,327 @@
+Make your choices soon, though, because the last day to order any item at the key chain tier or above is <strong>December 31st!</strong>
+</p>
+<p>
+Here's a preview of some of the cool stuff you can get if you order before the deadline:
```
Should we make these depend on whether they've already gotten them?  Like, if someone has already bought a tshirt, maybe there's no need to advertise the tshirt.

EliAndrewC replied:
You could do that by saying something like

```
{% if attendee.amount_extra < c.SHIRT_LEVEL %}
   ...
{% endif %}
```

EliAndrewC commented on magprime/automated_emails.py:
```diff
@@ -65,3 +65,15 @@ def __init__(self, event):
+AutomatedEmail(Attendee, 'Last Chance for MAGFest 2016 bonus swag!', 'attendee_swag_promo.html',
+               lambda a: a.can_spam and
+                         a.paid == c.HAS_PAID and
+                         days_ago(3, a.registered))
```
Your 1st and 3rd line look good.  As for the paid, it should be something like

```
(a.paid == c.HAS_PAID or a.group and a.group.amount_paid)
```

We might also want a condition where we check whether they're under a certain donation level already.  Like, if someone is already a Season Supporter, we probably don't need to send them this email since there's not much above that.  At the very least, if someone has kicked in at the absolute highest level, we should probably not send this email.

nickthenewbie1 replied:
I'm fine either way, but in the interest of time, I just want to get this
thing out the door.

--
Nicholas Marinelli
Chief Operating Officer
MAGFest
http://magfest.org
On Dec 22, 2015 6:42 PM, "Eli Courtwright" <notifications@github.com> wrote:

> In magprime/automated_emails.py
> <https://github.com/magfest/magprime/pull/50#discussion_r48310979>:
>
> > @@ -65,3 +65,15 @@ def __init__(self, event):
> >
> >  StopsEmail('MAGFest Dept Checklist Introduction', 'dept_checklist_intro.txt',
> >             lambda a: a.is_single_dept_head and a.admin_account)
> > +
> > +"""
> > +TODO: change the condition of this to be when:
> > +- can_spam == True
> > +- today's date < 3 days before the deadline in the email of Dec 31st 2015
> > +- attendee registered more than 3 days ago (wait to send them this email til 3 days after they have registered)
> > +- attendee paid or part of a paid group (but not part of an unpaid group, like a waitlisted dealer)
> > +"""
> > +AutomatedEmail(Attendee, 'Last Chance for MAGFest 2016 bonus swag!', 'attendee_swag_promo.html',
> > +               lambda a: a.can_spam and
> > +                         a.paid == c.HAS_PAID and
> > +                         days_ago(3, a.registered))
>
> Your 1st and 3rd line look good. As for the paid, it should be something
> like
>
> (a.paid == c.HAS_PAID or a.group and a.group.amount_paid)
>
> We might also want a condition where we check whether they're under a
> certain donation level already. Like, if someone is already a Season
> Supporter, we probably don't need to send them this email since there's not
> much above that. At the very least, if someone has kicked in at the
> absolute highest level, we should probably not send this email.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/magfest/magprime/pull/50/files#r48310979>.
>

binary1230 commented:
ok @EliAndrewC finished revisions.  we didn't put any limiting stuff in terms of "if they already paid don't show them", because it seems like a user would understand that this is sort of a promotional email pushed out to everyone.

I updated the lamba conditions and we tested it a lot locally, so pretty confident in this.  the new addition is adding in c.NEED_NOT_PAY badges with the intention of sending emails to staffers/guests/bands/etc.    I'm trying to think if there's any weird side effects of this and at worst it just means someone gets an extra email I guess.

Tested the email conditions a ton , and tested the HTML output in mailinator and gmail using real AWS keys on my local instance, so pretty happy with the formatting.

Fixed a couple typos.

Lemme know what you think.

EliAndrewC commented:
This looks good, though I still think we should add in ``a.amount_extra < c.SUPPORTER_LEVEL`` or at least ``a.amount_extra < c.SEASON_LEVEL`` or else we risk annoying people who already gave us money by advertising things they already bought.  Not a dealbreaker though, so feel free to merge this without that addition if you disagree.

A backlog issue in this repo for adding conditionals to the email body would be nice, so we can take the time to do a more customized job next year, but that's also not something that needs to hold this up.

My final suggestion is to test this out on mailinator from the staging server after it gets merged, just to make sure that the email renders in a reasonable way before it gets blasted out to thousand of people.  You can also test from GMail if you set the value of ``c.DEVELOPER_EMAIL`` in ``development.ini``, because that user is ALSO able to receive emails from the staging server for just this sort of reason.

nickthenewbie1 commented:
We did a mailinator test, and it rendered not so perfectly in mailinator,
but forwarded to gmail it looked just fine. This may be impetus to look
into more madmimi integration next year if we want to more regularly send
pretty e-mails.

I understand not wanting to annoy the supporters, though this e-mail is
double duty of advertising the swag, and giving people who already ordered
it a preview of what to expect. You're right that we can revisit the
conditionals next year.

--
Nicholas Marinelli
Chief Operating Officer
MAGFest
http://magfest.org

On Wed, Dec 23, 2015 at 5:06 PM, Eli Courtwright <notifications@github.com>
wrote:

> This looks good, though I still think we should add in a.amount_extra <
> c.SUPPORTER_LEVEL or at least a.amount_extra < c.SEASON_LEVEL or else we
> risk annoying people who already gave us money by advertising things they
> already bought. Not a dealbreaker though, so feel free to merge this
> without that addition if you disagree.
>
> A backlog issue in this repo for adding conditionals to the email body
> would be nice, so we can take the time to do a more customized job next
> year, but that's also not something that needs to hold this up.
>
> My final suggestion is to test this out on mailinator from the staging
> server after it gets merged, just to make sure that the email renders in a
> reasonable way before it gets blasted out to thousand of people. You can
> also test from GMail if you set the value of c.DEVELOPER_EMAIL in
> development.ini, because that user is ALSO able to receive emails from
> the staging server for just this sort of reason.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/magfest/magprime/pull/50#issuecomment-166999647>.
>

nickthenewbie1 commented:
Are we able to get this merged and deployed today?

--
Nicholas Marinelli
Chief Operating Officer
MAGFest
http://magfest.org

On Wed, Dec 23, 2015 at 5:13 PM, Nick Marinelli <nick@magfest.org> wrote:

> We did a mailinator test, and it rendered not so perfectly in mailinator,
> but forwarded to gmail it looked just fine. This may be impetus to look
> into more madmimi integration next year if we want to more regularly send
> pretty e-mails.
>
> I understand not wanting to annoy the supporters, though this e-mail is
> double duty of advertising the swag, and giving people who already ordered
> it a preview of what to expect. You're right that we can revisit the
> conditionals next year.
>
> --
> Nicholas Marinelli
> Chief Operating Officer
> MAGFest
> http://magfest.org
>
> On Wed, Dec 23, 2015 at 5:06 PM, Eli Courtwright <notifications@github.com
> > wrote:
>
>> This looks good, though I still think we should add in a.amount_extra <
>> c.SUPPORTER_LEVEL or at least a.amount_extra < c.SEASON_LEVEL or else we
>> risk annoying people who already gave us money by advertising things they
>> already bought. Not a dealbreaker though, so feel free to merge this
>> without that addition if you disagree.
>>
>> A backlog issue in this repo for adding conditionals to the email body
>> would be nice, so we can take the time to do a more customized job next
>> year, but that's also not something that needs to hold this up.
>>
>> My final suggestion is to test this out on mailinator from the staging
>> server after it gets merged, just to make sure that the email renders in a
>> reasonable way before it gets blasted out to thousand of people. You can
>> also test from GMail if you set the value of c.DEVELOPER_EMAIL in
>> development.ini, because that user is ALSO able to receive emails from
>> the staging server for just this sort of reason.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/magfest/magprime/pull/50#issuecomment-166999647>.
>>
>
>

kitsuta commented:
Because Nick and I discussed the business logic behind the last commit I added, I'm comfortable merging this with that change.
